pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -95,9 +95,13 @@ func Resolve(tpl integration.Config, svc listeners.Service) (integration.Config,
 		return resolvedConfig, errors.New("unable to resolve, service not ready")
 	}
 
-	tags, err := svc.GetTags()
-	if err != nil {
-		return resolvedConfig, err
+	tags := []string{}
```

Comment:
Instead of having a slice later processed which could be empty in some cases, it may have been cleaner to have:
```
	for i := 0; i < len(resolvedConfig.Instances); i++ {
		err = resolvedConfig.Instances[i].MergeAdditionalTags(tags)
		if err != nil {
			return resolvedConfig, err
		}
	}
```
runned only if `!tpl.IgnoreListenerTags` (moving the `svc.GetTags()` also there obv).

Something like:
```
	if !tpl.IgnoreListenerTags {
		tags, err = svc.GetTags()
		if err != nil {
			return resolvedConfig, err
		}
		for i := 0; i < len(resolvedConfig.Instances); i++ {
			err = resolvedConfig.Instances[i].MergeAdditionalTags(tags)
			if err != nil {
				return resolvedConfig, err
			}
		}
	}
```

This way, all the logic around tags is in the same block in this function. WDYT?

Reply:
good idea

Reply:
@remeh let me know if you are ok with my change.